caffe2/core/nomnigraph/include/nomnigraph/Graph/Graph.h

```diff
@@ -248,17 +272,15 @@ class Graph {
     const auto outEdges = old->getOutEdges();
 
     for (const auto& inEdge : inEdges) {
-      createEdge(inEdge->tail(), newTail);
-    }
-    for (const auto& inEdge : inEdges) {
-      deleteEdge(inEdge);
+      auto producer = inEdge->tail();
+      producer->replaceOutEdge(inEdge, createFloatingEdge(producer, newTail));
+      deleteEdge(inEdge, false);
```

Comment:
Why do you remove the old edge and create a new edge to replace it? Wouldn't updating the tail of the old edge be enough?